cmd/cluster-node-tuning-operator/main.go

```diff
@@ -219,48 +219,65 @@ func removePerformanceOLMOperator(cfg *rest.Config) error {
 		return err
 	}
 	csvSelector.Add(*req)
-	if err := k8sclient.List(context.TODO(), csvs, client.MatchingLabelsSelector{Selector: csvSelector}); err != nil {
-		if !errors.IsNotFound(err) {
-			return err
-		}
+
+	//REVIEW - should this be an input parameter? or something configurable?
+	paginationLimit := 50
+	opts := []client.ListOption{
+		client.MatchingLabelsSelector{Selector: csvSelector},
+		client.Limit(paginationLimit),
 	}
-	for i := range csvs.Items {
-		csv := &csvs.Items[i]
-		deploymentSpecs := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs
-		if deploymentSpecs != nil {
-			for _, deployment := range deploymentSpecs {
-				if deployment.Name == "performance-operator" {
-					performanceOperatorCSVs = append(performanceOperatorCSVs, *csv)
-					break
+	for {
+		if err := k8sclient.List(context.TODO(), csvs, opts...); err != nil {
+			if !errors.IsNotFound(err) {
+				return err
+			}
+		}
+		continueToken := csvs.GetContinue()
```

Comment:
can this panic / fail ?

Reply:
mm I think it should not panic. 
error from `List` is checked just above and the marker to know if there is no more elements is an empty continuation token so ...